### PR TITLE
Generic SWxSOC Integration

### PR DESCRIPTION
**Pull request by Alrobbertz**
The goal of this PR is to separate all non-HERMES-specific functionality of the `hermes_core` package to a new generic multi-mission core package. This new multi-mission core package can then be used to share common functionality including CDF file writing, metadata derivations, and schema validation, with other Missions code bases. 

This was partially inspired by discussions at DASH, at the PyHC presentation, at DASH, and my own browsing of other missions that would need to use functionality similar to that in `hermes_core`. 

The main impacts are:
- I/O File Handling completely migrated to generic package
- ISTP required schema derivation and validation migrated to generic package
    - Hermes-specific schema updated here
- Data class functionality moved to generic package

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Physics",
 ]
 dependencies = [
+  'swxsoc @ git+https://github.com/alrobbertz/swxsoc@integration_fixes',
```
Comment (Alrobbertz) [inline pyproject.toml:30]: NOTE: Temp branch pinned to test changes here before they're merged in https://github.com/swxsoc/swxsoc/pull/12

Comment (Alrobbertz) [reply pyproject.toml]: Updated!

File: docs/user-guide/customization.rst [hunk 2]
```diff
@@ -4,67 +4,25 @@
...
-  /home/vscode/.config/hermes_core/
-
-If you do not use our developer environment, you can run the following code to see where to place it on your specific machine as well:
-
-.. doctest::
-
-  >>> from hermes_core import util
-  >>> print(util.config._get_user_configdir())
-  /home/vscode/.config/hermes_core
+The `hermes_core` package uses an inherited configuration file from the `~swxsoc.util.config` module.
```
Comment (dbarrous) [inline docs/user-guide/customization.rst:10]: Is it possible to link to the swxsoc docs directly? https://hermes-core--122.org.readthedocs.build/en/122/user-guide/customization.html It looks like it doesn't link via this build

File: docs/user-guide/reading_writing_data.rst [hunk 3]
```diff
@@ -573,12 +573,12 @@ From this you can validate and distribute your CDF file.
 Validating a CDF File
 =====================
 
-The :py:class:`~hermes_core.timedata.HermesData` uses the `~spacepy.pycdf.istp` module for CDF validation, in addition to custom
-tests for additional metadata. A CDF file can be validated using the :py:func:`~hermes_core.util.validation.validate` method
-and by passing, as a parameter, the full :py:class:`~pathlib.Path` to the CDF file to be validated::
+HERMES supports validation of CDF files to ensure that they are compliant with the ISTP standards.
+The :py:class:`~hermes_core.timedata.HermesData` delegates to the `~swxsoc.util.validation` module for CDF validation.
```
Comment (dbarrous) [inline docs/user-guide/reading_writing_data.rst:577]: Same here, I think you will need to link to the docs page for `swxsoc.util.validation`

File: pyproject.toml [hunk 4]
```diff
@@ -27,6 +27,7 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Physics",
 ]
 dependencies = [
+  'swxsoc @ git+https://github.com/swxsoc/swxsoc',
```
Comment (dbarrous) [inline pyproject.toml:30]:
```suggestion
  'swxsoc @ git+https://github.com/swxsoc/swxsoc@main',
```
Quick change just to ensure it has the same dependency as when referenced in the pipeline

**Review by dbarrous — CHANGES_REQUESTED** [review]
Other than the changes below, this looks good! Will approve when fixed

File: docs/user-guide/reading_writing_data.rst (hunk 3 above)
Comment (Alrobbertz) [reply docs/user-guide/reading_writing_data.rst:577]: ^ It looks like this one does link to the `swxsoc` docs. 
https://hermes-core--122.org.readthedocs.build/en/122/user-guide/reading_writing_data.html#validating-a-cdf-file

File: docs/user-guide/customization.rst (hunk 2 above)
Comment (Alrobbertz) [reply docs/user-guide/customization.rst:10]: It looks like the hermes-core readthedocs can link to the swxsoc docs fine, just that the `config` module is not part of the `swxsoc` api docs. I'm going to make a PR there to update the API docs and then this link should work.

Comment (Alrobbertz) [reply docs/user-guide/customization.rst:10]: https://github.com/swxsoc/swxsoc/pull/16

**Review by dbarrous — APPROVED** [review]
LGTM after the docs changes were made. Thanks!

**Pull request merged**